src/rpc/request.cpp

```diff
@@ -37,21 +37,28 @@ UniValue JSONRPCRequestObj(const std::string& strMethod, const UniValue& params,
     return request;
 }
 
-UniValue JSONRPCReplyObj(const UniValue& result, const UniValue& error, const UniValue& id)
+UniValue JSONRPCReplyObj(const UniValue& result, const UniValue& error, const UniValue& id, JSONVersion json_version)
 {
     UniValue reply(UniValue::VOBJ);
-    if (!error.isNull())
-        reply.pushKV("result", NullUniValue);
-    else
+    // Add JSON-RPC version number field in v2 only.
+    if (json_version == JSONVersion::JSON_2_0) reply.pushKV("jsonrpc", "2.0");
+
+    // Add both result and error fields in v1, even though one will be null.
+    // Omit the null field in v2.
+    if (error.isNull()) {
         reply.pushKV("result", result);
-    reply.pushKV("error", error);
+        if (json_version == JSONVersion::JSON_1_BTC) reply.pushKV("error", NullUniValue);
+    } else {
+        reply.pushKV("error", error);
+        if (json_version == JSONVersion::JSON_1_BTC) reply.pushKV("result", NullUniValue);
```

Comment:
In commit "rpc: make error and result mutually exclusive in JSON-RPC 2.0 replies" (c9196a2e5b747753e0d5ee59e1bcb987943eedbf)

I think these two lines should be switched to always return the result field before the error field. This would make responses look more consistent and avoid changing previous behavior.